src/hardware/pounder/mod.rs

```diff
@@ -91,40 +93,84 @@ impl From<Channel> for GpioPin {
     }
 }
 
-#[derive(Serialize, Deserialize, Copy, Clone, Debug)]
-pub struct DdsChannelState {
-    pub phase_offset: f32,
+#[derive(Serialize, Deserialize, Copy, Clone, Debug, Miniconf)]
+pub struct DdsChannelConfig {
     pub frequency: f32,
+    pub phase_offset: f32,
     pub amplitude: f32,
-    pub enabled: bool,
 }
 
-#[derive(Serialize, Deserialize, Copy, Clone, Debug)]
-pub struct ChannelState {
-    pub parameters: DdsChannelState,
-    pub attenuation: f32,
+impl Default for DdsChannelConfig {
+    fn default() -> Self {
+        Self {
+            frequency: 0.0,
+            phase_offset: 0.0,
+            amplitude: 0.0,
+        }
+    }
 }
 
-#[derive(Serialize, Deserialize, Copy, Clone, Debug)]
-pub struct InputChannelState {
-    pub attenuation: f32,
-    pub power: f32,
-    pub mixer: DdsChannelState,
+pub struct ProfileWrapper(pub Profile);
+
+impl From<(ClockConfig, ChannelConfig)> for ProfileWrapper {
+    fn from((clocking, channel): (ClockConfig, ChannelConfig)) -> Self {
+        let system_clock_frequency =
+            clocking.reference_clock * clocking.multiplier as f32;
+        ProfileWrapper(Profile {
+            ftw: frequency_to_ftw(
+                channel.dds.frequency,
+                system_clock_frequency,
+            )
+            .unwrap(),
+            pow: phase_to_pow(channel.dds.phase_offset).unwrap(),
+            acr: amplitude_to_acr(channel.dds.amplitude).unwrap(),
+        })
+    }
 }
 
-#[derive(Serialize, Deserialize, Copy, Clone, Debug)]
-pub struct OutputChannelState {
+#[derive(Serialize, Deserialize, Copy, Clone, Debug, Miniconf)]
+pub struct ChannelConfig {
+    #[miniconf(defer)]
+    pub dds: DdsChannelConfig,
     pub attenuation: f32,
-    pub channel: DdsChannelState,
 }
 
-#[derive(Serialize, Deserialize, Copy, Clone, Debug)]
-pub struct DdsClockConfig {
+impl Default for ChannelConfig {
+    fn default() -> Self {
+        ChannelConfig {
+            dds: DdsChannelConfig::default(),
+            attenuation: 31.5,
+        }
+    }
+}
+
+#[derive(Serialize, Deserialize, Copy, Clone, Debug, PartialEq, Miniconf)]
+pub struct ClockConfig {
     pub multiplier: u8,
     pub reference_clock: f32,
     pub external_clock: bool,
 }
 
+impl Default for ClockConfig {
+    fn default() -> Self {
+        Self {
+            multiplier: super::design_parameters::DDS_MULTIPLIER,
+            reference_clock: super::design_parameters::DDS_REF_CLK.to_Hz()
+                as f32,
+            external_clock: false,
+        }
+    }
+}
+
+#[derive(Copy, Clone, Debug, Default, Miniconf)]
+pub struct PounderConfig {
+    pub clock: ClockConfig,
+    #[miniconf(defer)]
+    pub in_ch: miniconf::Array<ChannelConfig, 2>,
```

Comment:
Just call them `in` and `out`, as this mimics how they are labeled on the front pannel. Alternatively, `channel_in` and `channel_out` would be fine.